File: app/src/main/java/com/sky/oa/adapter/itemtouch/ItemTouchHelperCallback.java
package com.sky.oa.adapter.itemtouch;//package com.sky.oa.adapter;
//
//import androidx.annotation.NonNull;
//import androidx.recyclerview.widget.ItemTouchHelper;
//import androidx.recyclerview.widget.RecyclerView;
//
///**
// * @Description:
// * @Author: 李彬
// * @CreateDate: 2022/4/13 4:04 下午
// * @Version: 1.0
// */
//public class ItemTouchHelperCallback extends ItemTouchHelper.Callback {
//    private ItemTouchHelperListener listener;
//
//    public ItemTouchHelperCallback(ItemTouchHelperListener listener) {
//        this.listener = listener;
//    }
//
//    @Override
//    public int getMovementFlags(@NonNull RecyclerView recyclerView, @NonNull RecyclerView.ViewHolder viewHolder) {
//        int dragFlags = ItemTouchHelper.UP | ItemTouchHelper.DOWN;
//        int swipe = 0;
//        return makeMovementFlags(dragFlags, swipe);
//    }
//
//    @Override
//    public boolean onMove(@NonNull RecyclerView recyclerView, @NonNull RecyclerView.ViewHolder viewHolder, @NonNull RecyclerView.ViewHolder target) {
//        listener.onItemMove(viewHolder.getLayoutPosition(), target.getLayoutPosition());
//        return false;
//    }
//
//    @Override
//    public void onSwiped(@NonNull RecyclerView.ViewHolder viewHolder, int direction) {
//
//    }
//
//    @Override
//    public boolean isItemViewSwipeEnabled() {
//        return false;
//    }
//}

File: app/src/main/java/com/sky/oa/adapter/itemtouch/ItemTouchHelperCallback.kt
package com.sky.oa.adapter.itemtouch

import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.ItemTouchHelper
import androidx.recyclerview.widget.RecyclerView

/**
 *
 * @Description:
 * @Author: 李彬
 * @CreateDate: 2022/4/13 3:11 下午
 * @Version: 1.0
 */
class ItemTouchHelperCallback(val callback: ItemTouchHelperListener) : ItemTouchHelper.Callback() {
    override fun getMovementFlags(recyclerView: RecyclerView, viewHolder: RecyclerView.ViewHolder): Int {
        val manager = recyclerView.layoutManager
        var dragFlags = 0
        var swipeFlags = 0
        if (manager is GridLayoutManager) {
            dragFlags = ItemTouchHelper.UP or ItemTouchHelper.DOWN or ItemTouchHelper.LEFT or ItemTouchHelper.RIGHT
            swipeFlags = 0
        } else {
            dragFlags = ItemTouchHelper.UP or ItemTouchHelper.DOWN
            swipeFlags = ItemTouchHelper.START or ItemTouchHelper.END
        }
        return makeMovementFlags(dragFlags, swipeFlags)
    }

    override fun onMove(recyclerView: RecyclerView, viewHolder: RecyclerView.ViewHolder, target: RecyclerView.ViewHolder): Boolean {
        if (viewHolder.itemViewType !== target.itemViewType) {
            return false
        }
        callback.onItemMove(viewHolder.adapterPosition, target.adapterPosition)
        return true
    }

    override fun onSwiped(viewHolder: RecyclerView.ViewHolder, direction: Int) {
    }

    override fun isLongPressDragEnabled(): Boolean {
        return super.isLongPressDragEnabled()
    }

    override fun isItemViewSwipeEnabled(): Boolean {
        return false
    }
}